tensorflow/core/platform/s3/s3_file_system.cc

```diff
@@ -653,6 +653,11 @@ Status S3FileSystem::RenameFile(const string& src, const string& target) {
   return Status::OK();
 }
 
+Status S3FileSystem::NeedsTempLocation(const string& path) {
+  return Status(tensorflow::error::FAILED_PRECONDITION,
```

Comment:
Can this return a status and a boolean (as a last-argument pointer)? Looks like it can fail to fetch the status, and asking callers to figure out which statuses to bubble up and which are normal seems like too much.

Reply:
+1. It is unfortunate that we return `Status::OK` for `True` and an error status otherwise in cases when both true/false return should not be treated as one being too erroneous.

`IsDirectory` also needs change, but I'm working on that, no need to do that here.

Reply:
Changing it to `Status FileSystem::HasAtomicMove(const string& path, bool* has_atomic_move)`. Let me know if this is not what you had in mind.

Reply:
It can only fail to fetch the status if the file system wasn't recognized.